iceoryx_hoofs/test/moduletests/test_cxx_filesystem.cpp

```diff
@@ -0,0 +1,159 @@
+    constexpr auto BASE_VALUE_RHS = static_cast<permsBaseType_t>(TEST_VALUE_RHS);
+
+    EXPECT_THAT(static_cast<permsBaseType_t>(TEST_VALUE_LHS ^ TEST_VALUE_RHS), Eq(BASE_VALUE_LHS ^ BASE_VALUE_RHS));
+}
+
+TEST(filesystem_test, permsSatisfiesBinaryComplementOperationCorrectly)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "c313cf42-4cf0-4836-95ff-129111a707b0");
+    constexpr perms TEST_VALUE = perms::owner_read;
+
+    constexpr auto BASE_VALUE = static_cast<permsBaseType_t>(TEST_VALUE);
+
+    EXPECT_THAT(static_cast<permsBaseType_t>(~TEST_VALUE), Eq(~BASE_VALUE));
+}
+
+TEST(filesystem_test, permsSatisfiesBinaryOrAssignmentOperationCorrectly)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d3611de8-f932-4485-9e64-6cd8af4526dc");
+    constexpr perms TEST_VALUE = perms::sticky_bit;
+    constexpr perms TEST_VALUE_RHS = perms::group_read;
+
+    auto sutBaseValue = static_cast<permsBaseType_t>(TEST_VALUE);
+    constexpr auto BASE_VALUE_RHS = static_cast<permsBaseType_t>(TEST_VALUE_RHS);
+
+    perms sut = TEST_VALUE;
+
+    EXPECT_THAT(static_cast<permsBaseType_t>(sut |= TEST_VALUE_RHS), Eq(sutBaseValue |= BASE_VALUE_RHS));
+}
+
+TEST(filesystem_test, permsSatisfiesBinaryAndAssignmentOperationCorrectly)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "03c139be-e3ec-477e-8598-5da93699ab75");
+    constexpr perms TEST_VALUE = perms::others_exec;
+    constexpr perms TEST_VALUE_RHS = perms::others_all;
+
+    auto sutBaseValue = static_cast<permsBaseType_t>(TEST_VALUE);
+    constexpr auto BASE_VALUE_RHS = static_cast<permsBaseType_t>(TEST_VALUE_RHS);
+
+    perms sut = TEST_VALUE;
+
+    EXPECT_THAT(static_cast<permsBaseType_t>(sut &= TEST_VALUE_RHS), Eq(sutBaseValue &= BASE_VALUE_RHS));
+}
+
+TEST(filesystem_test, permsSatisfiesBinaryExclusiveOrAssignmentOperationCorrectly)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dae75205-a635-4535-8e8d-05541bb05b60");
+    constexpr perms TEST_VALUE = perms::none;
+    constexpr perms TEST_VALUE_RHS = perms::owner_all;
+
+    auto sutBaseValue = static_cast<permsBaseType_t>(TEST_VALUE);
+    constexpr auto BASE_VALUE_RHS = static_cast<permsBaseType_t>(TEST_VALUE_RHS);
+
+    perms sut = TEST_VALUE;
+
+    EXPECT_THAT(static_cast<permsBaseType_t>(sut ^= TEST_VALUE_RHS), Eq(sutBaseValue ^= BASE_VALUE_RHS));
+}
+
+TEST(filesystem_test, permsWhenEverythingIsSetTheOutputPrintsEverything)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2bb4931f-6ef9-4089-88a1-bf263a931559");
+    Logger_Mock loggerMock;
+    {
+        auto logStream = iox::log::LogStream(loggerMock);
+        logStream << perms::mask;
+    }
+
+    ASSERT_THAT(loggerMock.m_logs.size(), Eq(1U));
+    EXPECT_THAT(loggerMock.m_logs[0].message,
+                Eq("owner: {read, write, execute},  group: {read, write, execute},  others: {read, write, execute},  "
+                   "special bits: {set_uid, set_git, sticky_bit}"));
+}
+
+TEST(filesystem_test, permsWhenNothingIsSetEveryEntryIsNone)
```

Comment:
Even though that the test do checks on the stream operator it is not reflected in the name.
Proposal: `streamOperatorPrintsCorrectlyWhenPermsAreUnset`